client/components/Main/CreateNewSubnavDropdown.tsx

```diff
@@ -16,14 +16,42 @@ interface IProps {
     privileges: {[key: string]: number};
     createEventFromTemplate(template: IEventTemplate): void;
     eventTemplates: Array<IEventTemplate>;
+    recentTemplatesId: Array<string>;
 }
 
 class CreateNewSubnavDropdownFn extends React.PureComponent<IProps> {
+    constructor(props: IProps) {
+        super(props);
+        this.getRecentTemplates = this.getRecentTemplates.bind(this);
+    }
+
+    getRecentTemplates() {
```

Comment:
This is better off as a selector, using `createSelector` ([example](https://github.com/superdesk/superdesk-planning/blob/develop/client/selectors/vocabs.ts#L29)). Then this can be passed directly as a prop.

Don't forget the ts types